File: gin-vue-admin-main/web/src/api/medicine/mtDrugTypeLevel.js
import service from '@/utils/request'

// 获取所有二级分类
export const getAllMtDrugTypeLevel = () => {
  return service({
    url: '/mtDrugTypeLevel/getMtDrugTypeLevelPublic',
    method: 'get'
  })
}

// 分页获取二级分类列表
export const getMtDrugTypeLevelList = (params) => {
  return service({
    url: '/mtDrugTypeLevel/getMtDrugTypeLevelList',
    method: 'get',
    params
  })
}

// 创建二级分类
export const createMtDrugTypeLevel = (data) => {
  return service({
    url: '/mtDrugTypeLevel/createMtDrugTypeLevel',
    method: 'post',
    data
  })
}

// 更新二级分类
export const updateMtDrugTypeLevel = (data) => {
  return service({
    url: '/mtDrugTypeLevel/updateMtDrugTypeLevel',
    method: 'put',
    data
  })
}

// 删除二级分类
export const deleteMtDrugTypeLevel = (params) => {
  return service({
    url: '/mtDrugTypeLevel/deleteMtDrugTypeLevel',
    method: 'delete',
    params
  })
}

// 批量删除二级分类
export const deleteMtDrugTypeLevelByIds = (params) => {
  return service({
    url: '/mtDrugTypeLevel/deleteMtDrugTypeLevelByIds',
    method: 'delete',
    params
  })
}

// 根据ID查询二级分类
export const findMtDrugTypeLevel = (params) => {
  return service({
    url: '/mtDrugTypeLevel/findMtDrugTypeLevel',
    method: 'get',
    params
  })
} 

File: gin-vue-admin-main/server/router/medicine/mt_drug_type_level.go
package medicine

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type MtDrugTypeLevelRouter struct{}

// InitMtDrugTypeLevelRouter 初始化 mtDrugTypeLevel表 路由信息
func (s *MtDrugTypeLevelRouter) InitMtDrugTypeLevelRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	mtDrugTypeLevelRouter := Router.Group("mtDrugTypeLevel").Use(middleware.OperationRecord())
	mtDrugTypeLevelRouterWithoutRecord := Router.Group("mtDrugTypeLevel")
	mtDrugTypeLevelRouterWithoutAuth := PublicRouter.Group("mtDrugTypeLevel")
	{
		mtDrugTypeLevelRouter.POST("createMtDrugTypeLevel", mtDrugTypeLevelApi.CreateMtDrugTypeLevel)             // 新建mtDrugTypeLevel表
		mtDrugTypeLevelRouter.DELETE("deleteMtDrugTypeLevel", mtDrugTypeLevelApi.DeleteMtDrugTypeLevel)           // 删除mtDrugTypeLevel表
		mtDrugTypeLevelRouter.DELETE("deleteMtDrugTypeLevelByIds", mtDrugTypeLevelApi.DeleteMtDrugTypeLevelByIds) // 批量删除mtDrugTypeLevel表
		mtDrugTypeLevelRouter.PUT("updateMtDrugTypeLevel", mtDrugTypeLevelApi.UpdateMtDrugTypeLevel)              // 更新mtDrugTypeLevel表
	}
	{
		mtDrugTypeLevelRouterWithoutRecord.GET("findMtDrugTypeLevel", mtDrugTypeLevelApi.FindMtDrugTypeLevel)       // 根据ID获取mtDrugTypeLevel表
		mtDrugTypeLevelRouterWithoutRecord.GET("getMtDrugTypeLevelList", mtDrugTypeLevelApi.GetMtDrugTypeLevelList) // 获取mtDrugTypeLevel表列表
		mtDrugTypeLevelRouterWithoutRecord.GET("getAllMtDrugTypeLevel", mtDrugTypeLevelApi.GetAllMtDrugTypeLevel)   // 获取所有二级分类
	}
	{
		mtDrugTypeLevelRouterWithoutAuth.GET("getMtDrugTypeLevelPublic", mtDrugTypeLevelApi.GetAllMtDrugTypeLevel) // mtDrugTypeLevel表开放接口
	}
}
